src/lsm/compaction.zig

```diff
@@ -246,9 +239,18 @@ pub fn CompactionType(
                 .tracer_slot = null,
                 .iterator_tracer_slot = null,
             };
+
+            compaction.iterator_a.reset();
+            compaction.iterator_b.reset();
+            compaction.table_builder.reset();
+
+            std.mem.set(u8, compaction.index_block_a, 0);
+            for (&compaction.data_blocks) |data_block| {
```

Comment:
Would this not also work?

`for (compaction.data_blocks) |*data_block| {`

Feels like we should indicate more directly that we want the reference to the data block. Or am I missing something w.r.t. the type itself?